File: models/implementations/gru_attention.js
/* GRU + graph attention. See matching Python module for training. */
(function(global){
'use strict';
const registry=typeof module!=='undefined'?require('../../shared/runtime/model-registry'):global.FraudAdapters;
const base=typeof module!=='undefined'?require('./_neural'):global.FraudNeuralPrimitives;
// Architecture: gru memory; attention readout; 2 graph layers.
const adapter={initialize:base.initialize,update:base.update,readout:base.readout,predict:base.predict};
registry.register('gru_attention',adapter);
if(typeof module!=='undefined')module.exports=adapter;
})(globalThis);

File: models/implementations/gru_attention.py
"""GRU + graph attention: explicit architecture using shared differentiable primitives."""
from ._neural import NeuralModel, train_design

MODEL_ID = "gru_attention"


class Model(NeuralModel):
    architecture = {
        "id": "gru_attention",
        "label": "GRU + graph attention",
        "hidden": 8,
        "memory": "gru",
        "readout": "attention",
        "layers": 2,
        "description": "Original design: GRU account memory plus two layers of learned temporal graph attention.",
    }


def train(args, data, labeled=None):
    return train_design(args, data, MODEL_ID, labeled)
